test cases/common/23 global arg/prog.c

```diff
@@ -10,6 +10,29 @@
 #error "Global argument not set"
 #endif
 
+#ifdef GLOBAL_NATIVE
+#ifndef ARG_NATIVE
+#error "Global is native but arg_native is not set."
+#endif
+#else
+#ifndef GLOBAL_CROSS
+#error "Neither global_cross nor glogal_native is set."
+#endif
+#ifdef ARG_CROSS
+#error "Global is native but arg_cross is set."
+#endif
+#endif
+
+#ifdef GLOBAL_CROSS
+#ifndef ARG_CROSS
+#error "Global is cross but arg_cross is not set."
+#endif
+#else
+#ifdef ARG_CROSS
+#error "Global is cross but arg_native is set."
+#endif
+#endif
```

Comment:
Would be good to add indentation here so that the if/else/endif nesting is clearer.